### DISCUSSION TITLE
How to replace HTMLLoader to display PDF inside air application

### DISCUSSION THREAD

**Discussion by zabdoul60460** [airsdk/Adobe-Runtime-Support#1888] [General]
With the latest version of adobe air the component HTMLLoader has been removing.
In my case, I used it to display pdf files inside my air application.
`
 if ( HTMLLoader.pdfCapability == HTMLPDFCapability.STATUS_OK )
    {
        if ( dgHisto.selectedIndex != -1 )
        {
            var urlCourTXT:String=dgHisto.selectedItem.courl;
             container.removeAllChildren();
            urlCourTXTServ=new urlManager().urlCourriers() + urlCourTXT;//+ "#statusbar=0";
            
            pdf.height=container.height-10;
            pdf.width=container.width-10;
            
            
            var request:URLRequest=new URLRequest( urlCourTXTServ );
           
            try
            {
                pdf.load( request );
            }
            catch (error:ArgumentError)
            {
                Alert.show( "An ArgumentError has occurred : error:ArgumentError" , "Attention", Alert.OK, this);
                
            }
            catch (error:SecurityError)
            {
                Alert.show( " error:SecurityError" , "Alert", Alert.OK, this);
            }
              
            var ui:UIComponent=new UIComponent();
            ui.addChild( pdf );
            container.addChild( ui );
        }
        
    }
`

In this sample, container is a mx:VBox with drag and drop event.
So I try to do the same with StageWebView but it doesn't work !

Could you help me to find a solution.
Thanks for helping.
Best regards

**Comment by 2jfw** [comment]
Some basic code would be:

	  public class WebView extends Sprite
		  {

			private var _webView : StageWebView;
	  
		    public function WebView()
		    {
			      stage.scaleMode = StageScaleMode.NO_SCALE;
			      
			      addEventListener(Event.ADDED_TO_STAGE,
					                     onAddedToStage);
		    }
		    
		    
		    private function onAddedToStage(event : Event) : void
		    {
			      removeEventListener(Event.ADDED_TO_STAGE,
					                        onAddedToStage);
			      
			      createWebView();
			      
			      _webView.loadURL("https://www.airsdk.dev");
		    }
		    
		    
		    private function createWebView() : void
		    {
			      _webView          = new StageWebView(true);
			      _webView.stage    = stage;
			      _webView.viewPort = new Rectangle(0,
					                                0,
					                                stage.stageWidth,
					                                stage.stageHeight);
		    }
	  }


In App-Descriptor:

    <windows>
        <UseWebView2>true</UseWebView2>
    </windows>

**Reply by 2jfw** [reply]
Oh, forgot to mention you need to have the Microsoft *WebView2* installed for this sample (https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section).
For displaying PDF it may be sufficient enough to not use it - you can then try to instantiate the `WebView` without the `true` `new StageWebView(false);` and do not add the `UseWebView2`-part in App Descriptor

**Reply by zabdoul60460** [reply]
Thanks for helping but how to include WebView on UI. Indeed, I try to add it on my vBox (container) but it didn't works.
Best regards.

**Reply by 2jfw** [reply]
There is no need to actually add anything - assigning the `stage` the `viewPort` and `x y width height` to the WebView is sufficient enough for it to appear.
